editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/create/ext/CreateFigureFeature.java

```diff
@@ -115,35 +125,80 @@ public CATEGORY_TYPE getCategoryType() {
 	public boolean canCreate(ICreateContext context) {
 		ContainerShape container = context.getTargetContainer();
 		Object containerBO = getBusinessObjectForPictogramElement(container);
+		
+		// creating on the Camel Context / Diagram
 		if (container instanceof Diagram) {
-			// we want to create something on the diagram rather than inside a container element - thats only allowed on camelContexts
 			CamelFile cf = ((CamelDesignEditor)getDiagramBehavior().getDiagramContainer()).getModel();
-			if (cf.getCamelContext() == null && cf.getChildElements().size()==0) {
+			// check if the CamelFile has a Camel Context
+			if (cf.isEmpty()) {
+				// if not, then we need to add one right away to prevent further problems
 				cf.addChildElement(new CamelContextElement(cf, null));
 			}
-			if (eip != null) return eip.getName().equalsIgnoreCase("route") || 
-				eip.getName().equalsIgnoreCase("rest") || eip.getName().equalsIgnoreCase("restConfiguration");
-			if (clazz != null) { 
+			
+			// if we got an EIP reference then we can check if thats allowed
+			// to be added to the camel context
+			if (eip != null) {
+				return NodeUtils.canBeAddedToCamelContextDirectly(eip);
+			// if we only have a class defined
+			} else if (clazz != null) { 
+				// then we need to instantiate it
 				Object obj = newInstance(clazz);
-				if (obj != null && obj instanceof AbstractCamelModelElement) return ((AbstractCamelModelElement)obj).getNodeTypeId().equalsIgnoreCase("route");
+				if (obj != null && 
+					obj instanceof AbstractCamelModelElement) {
+					// and check if the element can be added to a context
+					return NodeUtils.canBeAddedToCamelContextDirectly((AbstractCamelModelElement)obj);
+				}
 				return false;
 			}
-		} else if (containerBO instanceof AbstractCamelModelElement && ((AbstractCamelModelElement)containerBO).getNodeTypeId().equalsIgnoreCase("choice")) {
+
+		// special handling for creating on choices (only one otherwise allowed)
+		} else if (	containerBO instanceof AbstractCamelModelElement && 
+					((AbstractCamelModelElement)containerBO).getNodeTypeId().equalsIgnoreCase("choice")) {
 			// only one otherwise per choice
 			AbstractCamelModelElement choice = (AbstractCamelModelElement)containerBO;
+			
 			if (this.eip != null) {
-				if (this.eip.getName().equalsIgnoreCase("otherwise") && choice.getParameter("otherwise") != null) return false;
-				return this.eip.getName().equalsIgnoreCase("when") || this.eip.getName().equalsIgnoreCase("otherwise");
+				if (this.eip.getName().equalsIgnoreCase("otherwise") && 
```

Comment:
for choice/when/otherwise (and some others), a constant is now available in org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement
As the code is modified close to it, we might improve this part